public/javascripts/Gallery/src/cards/CardBucket.js

```diff
@@ -62,13 +62,28 @@ function CardBucket(inputCards) {
     function copy() {
         return new CardBucket([...bucket]);
     }
+    
+    /**
+     * Gets the card that has the matching imageId
+     * 
+     * @param {String} imageId the id to search for
+     * @returns {Card} The card in the card bucket that contains the imageId
+     */
+    function findCardByImageId(imageId) {
+        for (let i = 0; i < bucket.length; i++) {
+            if (bucket[i].getImageId() == imageId) {
+                return bucket[i]
```

Comment:
Another thing to check everywhere: you're missing semicolons in a lot of places with this new JavaScript code.

Reply:
still seeing tons of missing semicolons. I'm fixing them as I see them, but you'll want to go through again with a fine-toothed comb later

Reply:
okay I ended up fixing them all. I just had a 200 line commit where over half was probably adding semicolons. just a PSA: with the power to "Resolve conversation" comes the great responsibility of making sure that things have actually been resolved :wink: